src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -292,6 +295,7 @@ private Handler<AsyncResult<KafkaConsumerRecords<K, V>>> pollHandler(RoutingCont
                 );
                 HttpUtils.sendResponse(routingContext, HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
                     BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+                span.finish(HttpResponseStatus.INTERNAL_SERVER_ERROR.code());
```

Comment:
My second doubt was that we have done abstraction over the span implementations right, so I used the finish method to set the error code but do we need to inject the RoutingContext into the span also as we are doing in case of successful poll?

Reply:
I don't think we need because anyway the span is going to be finished.